Bottom: 623a475191efa43ebea39d2b09a61e581a20047d
Top:    df7200845506d8b64934397a1a337c72e520ccf9
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-12 17:50:50 +0100

Increase initial heap size from 32 to 256

In complex boards with lots of polygons, we were realloc'ing up
to this kind of number anyway, in x2 size steps. Save all the
memcpy'ing by making the heap larger to start with.

My test board still has some requirement for heaps >256, <512,
but the frequency is low.



---

diff --git a/src/heap.c b/src/heap.c
index 39dcbe4..4959b58 100644
--- a/src/heap.c
+++ b/src/heap.c
@@ -165,7 +165,7 @@ heap_insert (heap_t * heap, cost_t cost, void *data)
     {
       heap->max *= 2;
       if (heap->max == 0)
-	heap->max = 32;		/* default initial heap size */
+	heap->max = 256;		/* default initial heap size */
       heap->element =
 	realloc (heap->element, heap->max * sizeof (*heap->element));
     }
